src/java.base/share/classes/jdk/internal/classfile/components/snippet-files/PackageSnippets.java

```diff
@@ -0,0 +1,212 @@
+        var classNode = ClassPrinter.toTree(classModel, ClassPrinter.Verbosity.TRACE_ALL);
+        assertContains(classNode, "method name", "myFooMethod");
+        assertContains(classNode, "field name", "myBarField");
+        assertContains(classNode, "inner class", "MyInnerFooClass");
+    }
+
+    void assertContains(ClassPrinter.Node node, ConstantDesc key, ConstantDesc value) {
+        if (!node.walk().anyMatch(n -> n instanceof ClassPrinter.LeafNode ln
+                               && ln.name().equals(key)
+                               && ln.value().equals(value))) {
+            node.toYaml(System.out::print);
+            throw new AssertionError("expected %s: %s".formatted(key, value));
+        }
+    }
+    // @end
+    @interface Test{}
+
+    void singleClassRemap(ClassModel... allMyClasses) {
+        // @start region="singleClassRemap"
+        var classRemapper = ClassRemapper.of(
+                Map.of(ClassDesc.of("Foo"), ClassDesc.of("Bar")));
+
+        for (var classModel : allMyClasses) {
+            byte[] newBytes = classRemapper.remapClass(classModel);
+
+        }
+        // @end
+    }
+
+    void allPackageRemap(ClassModel... allMyClasses) {
+        // @start region="allPackageRemap"
+        var classRemapper = ClassRemapper.of(cd ->
+                ClassDesc.ofDescriptor(cd.descriptorString().replace("Lcom/oldpackage/", "Lcom/newpackage/")));
+
+        for (var classModel : allMyClasses) {
+            byte[] newBytes = classRemapper.remapClass(classModel);
+
+        }
+        // @end
+    }
+
+    void codeLocalsShifting(ClassModel classModel) {
+        // @start region="codeLocalsShifting"
+        byte[] newBytes = classModel.transform((classBuilder, classElement) -> {
+                if (classElement instanceof MethodModel method)
+                    classBuilder.transformMethod(method,
+                            MethodTransform.transformingCode(
+                                    CodeLocalsShifter.of(method.flags(), method.methodTypeSymbol())));
+                else
+                    classBuilder.accept(classElement);
+            });
+        // @end
+    }
+
+    void codeRelabeling(ClassModel classModel) {
+        // @start region="codeRelabeling"
+        byte[] newBytes = classModel.transform(
+                ClassTransform.transformingMethodBodies(
+                        CodeTransform.ofStateful(CodeRelabeler::of)));
+        // @end
+    }
+
+    // @start region="classInstrumentation"
+    byte[] classInstrumentation(ClassModel target, ClassModel instrumentor, Predicate<MethodModel> instrumentedMethodsFilter) {
+        var instrumentorCodeMap = instrumentor.methods().stream()
+                                              .filter(instrumentedMethodsFilter)
+                                              .collect(Collectors.toMap(mm -> mm.methodName().stringValue() + mm.methodType().stringValue(), mm -> mm.code().orElse(null)));
```

Comment:
Should we be filtering out abstract methods before the `collect` and/or replace with:
```
mm -> mm.code().orElseThrow()
```
?

Reply:
Abstract methods should not be selected for instrumentation, so `orElseThrow()` is a good idea.
Fixed, thanks.